blog/_posts/2022-01-03-arm-wrestling-in-your-dev-env.md

```diff
@@ -0,0 +1,103 @@
+---
+slug: "arm-wrestling-in-dev"
+date: 2022-01-04
+author: nick
+layout: blog
+title: "ARM Wrestling the Right Processor in Dev"
+subtitle: "Ways Tilt can help you set the right platform for your images"
+description: "Ways Tilt can help you set the right platform for your images"
+image: "/assets/images/arm-wrestling-in-dev/cover.jpg"
+image_caption: "The M1 high-throughput bus in action. Wah wah. Courtesy of <a href='https://commons.wikimedia.org/wiki/File:MTA_NYC_Bus_M1_bus_at_Broadway_%26_8th_St.jpg'>Mtattrain on Wikimedia.org</a>."
+tags:
+  - docker
+  - arm
+---
+
+Maybe you got a fancy new MacOS M1 laptop and now you're getting cryptic errors
+when you build your app.
+
+Or maybe you're trying out [EC2
+Graviton](https://aws.amazon.com/ec2/graviton/) machines in your dev cluster,
+but now none of your images work right.
+
+Welcome to the world of ARM processors! The x86 monoculture has been broken!
+
+This is good for the ecosystem in general. But now your dev environment has to
+handle the case where your host CPU is different from the target CPU.
+
+Tilt has shipped a few small tweaks recently that we hope will help build for
+the right CPU without having to think about it.
+
+## Setting Platform Automatically
+
+If you're using Tilt to build Docker images with `docker_build`,
+we'll now automatically read the architecture off the cluster
+and use that in the Docker build.
+
+In the logs, you'll see a line like this:
+
+```
+STEP 1/5 — Building Dockerfile: [tilt-site]
+Building Dockerfile for platform linux/amd64:
+```
+
+We've detected that the cluster is `amd64` and will use this instead of your
+host architecture.
+
+## Setting Platform Manually
+
+Maybe you want to force a particular architecture, or build multi-platform images in dev.
+
+You can override the platform manually by:
+
+- Setting the `DOCKER_DEFAULT_PLATFORM` [env var](https://docs.docker.com/engine/reference/commandline/cli/#environment-variables)
+
+```bash
+export DOCKER_DEFAULT_PLATFORM="linux/amd64,linux/arm64"
+tilt up
+```
+
+- Setting the `platform` argument to `docker_build` in your Tiltfile. 
+
+```python
+docker_build('fe', '.', platform='linux/amd64,linux/arm64')
+```
+
+## Wiring it Yourself
+
+One of our goals with Tilt is to expose every part of your dev environment
+as a Kubernetes-style API, so that you can use it in your own scripts
+to make decisions.
```

Comment:
```suggestion
to make decisions in an automated manner.
```
Not too sure about the wording here, but I would like to add the "and you can thus automate everything!" idea to it

Reply:
done